modules/configuration/src/main/java/org/apache/ignite/internal/configuration/tree/ConverterToMapVisitor.java

```diff
@@ -107,15 +106,22 @@ public Object visitInnerNode(Field field, String key, InnerNode node) {
 
         deque.pop();
 
+        String injectedValueFieldName = node.injectedValueFieldName();
+
+        // If configuration contains an injected value, we need to take the value of the synthetic key and use it as the key for
+        // the injected value. For example, instead of ["syntheticKey": "foo", "injectedValue": "bar"], we will have ["foo": "bar"].
+        if (injectedValueFieldName != null) {
+            innerMap.put(key, innerMap.remove(injectedValueFieldName));
```

Comment:
Something's wrong here, I feel it.
Do we generate `[{foo1 : bar1}, {foo2 : bar2}]` instead of `{foo1 : bar1, foo2 : bar2}`?
Is this what happens? That's a weird syntax, please fix it, I don't think we should support it